New: Add switch-final-break rule (#12094)
eslint/eslint: docs/rules/switch-final-break.md

> # Checks whether the final clause of a switch statement ends in unnecessary `break;`. (switch-final-break)

Reviewer: Can we add a few sentences that describe why one might want to use this rule?

Examples of **correct** code:

```js
switch (x) {
  case 0: {
    foo();
  }
}
```

Examples of **incorrect** code:

```js
switch (x) {
  case 0: {
    foo();
    break;
  }
}
```

## Options

> Can get `never` (the default) or `always` (always ends with `break;`):

Reviewer: I think we can improve the clarity of this line by formatting it similarly to this section if the  [`rest-spread-spacing` documentation](https://eslint.org/docs/rules/rest-spread-spacing#options).

```json
{
  "switch-final-break": ["error", "always"]
}

> ```

Reviewer: Can we add a “When not to use this rule” section? See [here](https://eslint.org/docs/rules/rest-spread-spacing#when-not-to-use-it) as an example.

> # Checks whether the final clause of a switch statement ends in unnecessary `break;`. (switch-final-break)

Reviewer: This should match what we have in the rule's `meta.description`: https://github.com/eslint/eslint/pull/12094/files#diff-cff99b58163f559f91380878b55965d9R21-R22

> The final clause of a switch statement can end with an unnecessary `break;` statement, which can lead to inconsistent within a codebase.

Reviewer: ```suggestion
The final clause of a switch statement can end with an unnecessary `break;` statement, which can lead to inconsistency within a codebase.
```

> This rule improve codebase consistency by preventing this last `break;` statement.

Reviewer: ```suggestion
This rule improves codebase consistency by requiring or disallowing this last `break;` statement.
```

## Options

### "never"

When using the default `"never"` option, an unnecessary `break;` statement at the end of a switch statement is not allowed.

```json
{
  "switch-final-break": ["error"]
}
```

or

```json
{
  "switch-final-break": ["error", "never"]
}
```

Examples of **incorrect** code for this rule with `"never"`:

```js
/*eslint switch-final-break: ["error", "never"]*/

switch (x) {
  case 0: {
    foo();
    break;
  }
}
```

Examples of **correct** code for this rule with `"never"`:

```js
/*eslint switch-final-break: ["error", "never"]*/

switch (x) {
  case 0: {
    foo();
  }
}
```

### "always"

When using the `"always"` option, a `break;` statement at the end of a switch statement is required.

```json
{
  "switch-final-break": ["error", "always"]
}
```

Examples of **incorrect** code for this rule with `"always"`:

```js
/*eslint switch-final-break: ["error", "always"]*/

switch (x) {
  case 0: {
    foo();
  }
}
```

Examples of **correct** code for this rule with `"always"`:

```js
/*eslint switch-final-break: ["error", "always"]*/

switch (x) {
  case 0: {
    foo();
    break;
  }
}
```

## When Not To Use It

> You can safely disable this rule if you do not care about enforcing consistent in the last clause of a switch statement regarding the use of the `break` statement.

Reviewer: ```suggestion
You can safely disable this rule if you do not care about the consistent use of  `break` statement in the last clause of a switch statement.
```
